streams/src/main/java/org/apache/kafka/streams/StoreQueryParams.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams;
+
+import org.apache.kafka.streams.state.QueryableStoreType;
+
+import java.util.Objects;
+
+/**
+ * Represents all the query options that a user can provide to state what kind of stores it is expecting.
+ * The options would be whether a user would want to enable/disable stale stores
+ * or whether it knows the list of partitions that it specifically wants to fetch.
+ * If this information is not provided the default behavior is to fetch the stores for all the partitions
+ * available on that instance for that particular store name.
+ * It contains a partition, which for a point queries can be populated from the {@link KeyQueryMetadata}.
+ */
+public class StoreQueryParams<T> {
+
+    private Integer partition;
+    private boolean staleStores;
+    private final String storeName;
+    private final QueryableStoreType<T> queryableStoreType;
+
+    private StoreQueryParams(final String storeName, final QueryableStoreType<T>  queryableStoreType) {
+        this.storeName = storeName;
+        this.queryableStoreType = queryableStoreType;
+    }
+
+    public static <T> StoreQueryParams<T> fromNameAndType(final String storeName,
+                                                          final QueryableStoreType<T>  queryableStoreType) {
+        return new StoreQueryParams<T>(storeName, queryableStoreType);
+    }
+
+    /**
+     * Set a specific partition that should be queried exclusively.
+     *
+     * @param partition   The specific integer partition to be fetched from the stores list by using {@link StoreQueryParams}.
+     *
+     * @return String storeName
+     */
+    public StoreQueryParams<T> withPartition(final Integer partition) {
+        final StoreQueryParams<T> storeQueryParams = StoreQueryParams.fromNameAndType(this.storeName(), this.queryableStoreType());
+        storeQueryParams.partition = partition;
+        storeQueryParams.staleStores = this.staleStores;
+        return storeQueryParams;
```

Comment:
It might be simpler to add a private constructor that allows to specify all 4 parameters:
```
return new StoreQueryParams<>(storeName, queryableStoreType, partition, staleStores);
```

Similar in other methods

Reply:
Done.